integrations/SD_Lon/fix_departments.py

```diff
@@ -464,23 +464,42 @@ def fix_or_create_branch(self, leaf_uuid, date):
         for unit in reversed(branch):
             self.fix_department_at_single_date(unit[1], date)
 
+
+    def sd_uuid_from_short_code(self, shortname, validity_date):
+        validity = {
+            'from_date': validity_date.strftime('%d.%m.%Y'),
+            'to_date': validity_date.strftime('%d.%m.%Y')
+        }
+        department = self.get_department(validity, shortname=shortname)[0]
+        return department["DepartmentUUIDIdentifier"]
+
+
+
     def _cli(self):
         """
         Command line interface to sync SD departent information to MO.
         """
         parser = argparse.ArgumentParser(description='Department updater')
-        parser.add_argument('--department-uuid', nargs=1, required=True,
+        parser.add_argument('--department-uuid', nargs="+", required=True,
                             metavar='UUID of the department to update')
         args = vars(parser.parse_args())
 
         today = datetime.datetime.today()
-        department_uuid = args.get('department_uuid')[0]
 
         # Use a future date to be sure that the unit exists in SD.
         fix_date = today + datetime.timedelta(weeks=80)
-        self.fix_or_create_branch(department_uuid, fix_date)
 
-        self.fix_NY_logic(department_uuid, today)
+        for some_id in args.get('department_uuid'):
+
+            if len(some_id) == 4:
+                department_uuid = self.sd_uuid_from_short_code(some_id, fix_date)
+            else:
+                department_uuid = some_id
+
+
+            self.fix_or_create_branch(department_uuid, fix_date)
+
+            self.fix_NY_logic(department_uuid, today)
```

Comment:
```suggestion
        def some_id_to_uuid(some_id):
            if len(some_id) == 4:
                return self.sd_uuid_from_short_code(some_id, fix_date)
            return some_id

        department_uuids = map(some_id_to_uuid, args.get('department_uuid'))
        for dapartment_uuid in department_uuids:
            self.fix_or_create_branch(department_uuid, fix_date)
            self.fix_NY_logic(department_uuid, today)
```
Jeg ville nok igen have lavet det her, som et `map`, således at oversættelsen bor i en funktion for sig selv som en logisk seperat enhed, og derudover bliver oversættelse og udførsel adskilt.

Reply:
Det ser meget bedre ud, men man forsøger her at slå alle sammen op inden man kører dem.

På den måde vil 

program 1234 1245 1236

opføre sig anderledes end

program 1234
program 1235
program 1236

Reply:
Løgn!

`map` er lazy. Så adfæren er den samme, da `some_id_to_uuid` vil blive kørt under `next` for iteratoren i `for`-løkken.

Reply:
hæhæ - nok om det. Den skal ud over rampen. Vil du trykke på knappen!